priv/blogs/2022/07-29-observing-elixir-with-lightstep.md

```diff
@@ -0,0 +1,222 @@
+
+### If high-cardinality data is requisite for observability, then we need two things:
+
+1. The ability to capture and export distributed traces of spans in our running system
+2. A tool for analyzing our traces to correlate issues with root causes
+
+We can achieve the above with [OpenTelemetry](https://opentelemetry.io/docs/instrumentation/erlang/) and an analysis tool like [Lightstep](https://www.honeycomb.io/) or [Honeycomb](https://www.honeycomb.io/).
+
+## Instrumenting your Elixir app with OpenTelemetry
+
+Let's instrument a brand new Elixir app with OpenTelemetry. For this guide, I'll assume you have a [Phoenix application](https://www.phoenixframework.org/) using [Elixir 1.13](https://elixir-lang.org/blog/2021/12/03/elixir-v1-13-0-released/) or newer.
+
+### Install Dependencies
+
+First we'll add the required dependencies into our `mix.exs` file for producing traces and spans with OpenTelemetry
+
+```elixir
+# mix.exs
+{:opentelemetry, "~> 1.0"},
+{:opentelemetry_api, "~> 1.0"},
+{:opentelemetry_exporter, "~> 1.0", only: :prod},
+```
+
+Additionally, you should install instrumentation libraries for any of the popular libraries and frameworks you may be using
+
+```elixir
+# mix.exs
+{:opentelemetry_ecto, "~> 1.0"},
+{:opentelemetry_liveview, "~> 1.0.0-rc.4"},
+{:opentelemetry_oban, "~> 0.2.0-rc.5"},
+{:opentelemetry_phoenix, "~> 1.0"},
+```
+
+Once you run `mix deps.get`, its time to instrument your application
+
+### Instrumenting libraries
+
+As we saw above, there are many packages above that can give you great tracing to the libraries you're already using. These libraries typically work by hooking into [telemetry](https://hexdocs.pm/telemetry/readme.html) events produced by these libraries and converting them into OpenTelemetry spans. Note that this is a point of confusion for many people, as the `telemetry` library is an BEAM-specific package for producing telemetry data inside a BEAM application, and is not directly related to OpenTelemetry. However, it is a useful mechanism for libraries, such as [opentelemetry_ecto](https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex#L42) to hook into for producing spans.
```

Comment:
Suggestion: Move the `:telemetry` vs. OpenTelemetry explanation into the introduction of the "Instrumenting your Elixir app with OpenTelemetry" section.

This will definitely be a source of confusion for Elixir folks, so I think it better to clarify before diving into OpenTelemetry.

shameless plug: Feel free to link to the [Phoenix Framework Telemetry guide](https://hexdocs.pm/phoenix/telemetry.html) as well for more Elixir- and Phoenix-specific telemetry info :)

Reply:
<img width="606" alt="Screen Shot 2022-08-03 at 8 52 48 AM" src="https://user-images.githubusercontent.com/1019721/182612478-04aca50c-c3cf-4761-b30f-719e4aa31ecc.png">

I dedicated this aside to explaining the difference

Reply:
Thank you for the suggestion!

Reply:
<img width="607" alt="Screen Shot 2022-08-03 at 9 03 46 AM" src="https://user-images.githubusercontent.com/1019721/182614720-6ea532e4-8e6d-4d70-892a-88670fe90e55.png">

Tweaked it again